### PR TITLE
Niad-3067: duplicated ObservationStatement entries for the same Observation

### PR DESCRIPTION
**Pull request by ole4ryb**
## What

ObservationStatement with "..." pattern at the end of pertinentAnnotation has been merged with one that has the full description of the same line.

## Why

There should be full description of pertinentAnnotation included instead of the truncated one.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Internal change (non-breaking change with no effect on the functionality affecting end users)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the [Changelog](/CHANGELOG.md) with details of my change in the UNRELEASED section if this change will affect end users
- [x] A corresponding change has been made to the [Mapping Documentation repository][mapping-docs]

[mapping-docs]: https://github.com/NHSDigital/patient-switching-adaptors-mapping-documentation

### CODE REVIEW

**Comment by adrianclay** [comment]
You sure this code meets Acceptance Criteria 2 of the ticket @ole4ryb / @MartinWheelerMT ?

**Comment by adrianclay** [comment]
I also wonder if over 600 lines of XML is necessary, or whether it could be cut down smaller.

**Comment by ole4ryb** [comment]
> I also wonder if over 600 lines of XML is necessary, or whether it could be cut down smaller.

There are under 600 lines of XML :) anyway, I will check and see if it can be reduced.

**Comment by ole4ryb** [comment]
> You sure this code meets Acceptance Criteria 2 of the ticket @ole4ryb / @MartinWheelerMT ?

Yes, I've run a test locally using Postman and see only one condition generated.

File: gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java [hunk 1]
```diff
@@ -454,6 +515,29 @@ private List<RCMRMT030101UKEhrComposition> getCompositionsContainingLinkSets(RCM
...
+        RCMRMT030101UKObservationStatement referencedObservationStatement) {
+
+        List<RCMRMT030101UKObservationStatement> observationStatements = ehrExtract.getComponent().stream()
+            .map(RCMRMT030101UKComponent::getEhrFolder)
+            .map(RCMRMT030101UKEhrFolder::getComponent)
+            .flatMap(Collection::stream)
+            .map(RCMRMT030101UKComponent3::getEhrComposition)
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .flatMap(CompoundStatementResourceExtractors::extractAllObservationStatements)
+            .toList();
```
Comment (adrianclay) [inline gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java:529]: Does this code satisfy this condition?

```
Be a child of the same EhrComposition
```

Comment (adrianclay) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java:529]: Also the condition:
```
The replaced ObservationStatement must have a single pertinentInformation / pertinentAnnotation which ends in ...
```

Comment (adrianclay) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java:529]: I think I can see that final condition being met on line 253, but the downside of this approach, is that the check happens after the `findFirst` call which means that we could be missing the `ObservationStatement` if the actual match isn't the first.

Comment (ole4ryb) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java:529]: I think we covered it with a test when the ordering is different and it works OK.

Comment (ole4ryb) [reply gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java:529]: > Also the condition:
> 
> ```
> The replaced ObservationStatement must have a single pertinentInformation / pertinentAnnotation which ends in ...
> ```

Yes, we've got it confirmed with tests.

**Pull request closed without merging**